multicore_runtime/wsgi_config.py

```diff
@@ -0,0 +1,216 @@
+from google.appengine.runtime import wsgi
+
+
+def get_module_config_filename():
+  """Returns the path of the module configuration file (e.g. 'app.yaml').
+
+  Returns:
+    The path of the module configuration file.
+
+  Raises:
+    KeyError: The MODULE_YAML_PATH environment variable is not set.
+  """
+  module_yaml_path = os.environ['MODULE_YAML_PATH']
+  logging.info('Using module_yaml_path from env: %s', module_yaml_path)
+  return module_yaml_path
+
+
+def get_module_config(filename):
+  """Returns the parsed module config."""
+  with open(filename) as f:
+    return appinfo_includes.Parse(f)
+
+
+def app_for_script(script):
+  """Returns the WSGI app specified in the input string, or None on failure."""
+  try:
+    app, unused_filename, err = wsgi.LoadObject(script)
+    if err:
+      raise err
+  except ImportError as e:
+    # Despite nominally returning an error object, LoadObject will sometimes
+    # just result in an exception. By raising the err object if it exists, we
+    # catch either.
+    logging.exception('Failed to import %s', script)
+    return None
+  else:
+    return app_wrapped_in_user_middleware(app)
+
+
+def app_wrapped_in_user_middleware(app):
+  """Returns the input WSGI app, wrapped in appengine_config middleware."""
+  add_middleware = get_add_middleware_from_appengine_config()
+  if add_middleware:
+    return add_middleware(app)
+  else:
+    return app
+
+
+def get_add_middleware_from_appengine_config():
+  """Tries to import appengine_config and return middleware; fails silently.
+
+  `env_config` is optionally part of GAE-compatible user code.
+  If the user chooses to define the `webapp_add_wsgi_middleware` function
+  there, then we will use it to wrap app scripts. This is used for e.g.
+  appstats. Many user apps do not have or need env_config, so failure
+  to import it or failure to find `webapp_add_wsgi_middleware` is not an
+  error.
+
+  Returns:
+    The appengine_config.webapp_add_wsgi_middleware function or None.
+  """
+  try:
+    import appengine_config  # pylint: disable=g-import-not-at-top
+    return getattr(appengine_config, 'webapp_add_wsgi_middleware', None)
+  except ImportError:
+    return None
+
+
+def static_app_for_handler(handler):
+  """Returns a WSGI app that serves static files as directed by the handler.
+
+  Args:
+    handler: An individual handler from appinfo_external.handlers
+      (appinfo.URLMap)
+
+  Returns:
+    A static file-serving WSGI app closed over the handler information.
+  """
+  url_re = handler.url  # handler.url is a regex, despite the name.
+  files = handler.static_files
+  upload = handler.upload
+  if not files:
+    if handler.static_dir:
+      # If static_files is not set, convert static_dir to static_files and also
+      # modify the url regex accordingly. See the appinfo.URLMap docstring for
+      # more information.
+      url_re = static_dir_url(handler)
+      files = '{dir}{backref}'.format(dir=handler.static_dir, backref=r'/\1')
```

Comment:
Why not incorporate the constant part into the format string itself?

```python
  files = r'{dir}/\1'.format(dir=handler.static_dir)
```

Yes, I see there's some value in giving it a name such as `backref` but perhaps you could accomplish the same by naming the variable `files_backref`?

Same for the line below and I think there's a handful more like it...